src/components/cl/doca_urom/cl_doca_urom.h

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2020, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_CL_DOCA_UROM_H_
+#define UCC_CL_DOCA_UROM_H_
+
+#include "components/cl/ucc_cl.h"
+#include "components/cl/ucc_cl_log.h"
+#include "components/tl/ucc_tl.h"
+#include "coll_score/ucc_coll_score.h"
+#include "utils/ucc_mpool.h"
+
+#include <doca_dev.h>
+#include <doca_urom.h>
+#include <doca_pe.h>
+#include <doca_ctx.h>
+#include <doca_buf.h>
+
+#include <worker_ucc.h>
+#include <common.h>
+#include <urom_common.h>
+#include <urom_ucc.h>
+
+#ifndef UCC_CL_DOCA_UROM_DEFAULT_SCORE
+#define UCC_CL_DOCA_UROM_DEFAULT_SCORE 100
+#endif
+
+#define UCC_CL_DOCA_UROM_ADDR_MAX_LEN  1024
+#define UCC_CL_DOCA_UROM_MAX_TEAMS     16
+
+typedef struct ucc_cl_doca_urom_iface {
+    ucc_cl_iface_t super;
+} ucc_cl_doca_urom_iface_t;
+
+// Extern iface should follow the pattern: ucc_cl_<cl_name>
+extern ucc_cl_doca_urom_iface_t ucc_cl_doca_urom;
+
+typedef struct ucc_cl_doca_urom_lib_config {
+    ucc_cl_lib_config_t super;
+    char *device;
```

Comment:
IIUC, device and allocation of ressource are usually set at the context level rather than at the lib level.

https://openucx.github.io/ucx/api/v1.8/html/group___u_c_p___c_o_n_t_e_x_t.html#ga0e40ddc46f0bbe868a13f6ab1b674f76

https://github.com/openucx/ucc/blob/b62b4723a7183496efc3792d90e7bfefcfd6d7ed/src/components/tl/ucp/tl_ucp.h#L117

https://github.com/openucx/ucc/blob/b62b4723a7183496efc3792d90e7bfefcfd6d7ed/src/components/tl/mlx5/tl_mlx5.h#L67